app/gui/src/project-view/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -119,17 +120,21 @@ class ExpressionTag {
     else return null
   }
 
-  static FromExpression(expression: string, label?: Opt<string>): ExpressionTag {
+  static FromExpression(
+    expression: string,
+    label?: Opt<string>,
+    icon?: Opt<string>,
+  ): ExpressionTag {
     const qn = tryQualifiedName(expression)
     if (qn.ok) {
       const projectPath = projectNames.parseProjectPath(qn.value)
       if (projectPath.ok) {
         const fromProjPath = ExpressionTag.FromProjectPath(projectPath.value, label)
         if (fromProjPath) return fromProjPath
       }
-      return new ExpressionTag(qn.value, label ?? qnLastSegment(qn.value))
+      return new ExpressionTag(qn.value, label ?? qnLastSegment(qn.value), icon)
```

Comment:
`isIconName` can be used to validate `icon` here (to satisfy the stricter type suggested for the constructor parameter)

Reply:
Changed